Neos.Neos/Resources/Private/Fusion/Prototypes/MenuDataItem.fusion

```diff
@@ -0,0 +1,44 @@
+prototype(Neos.Neos:MenuDataItem) < prototype(Neos.Fusion:Component) {
+  currentDocumentNode = ${documentNode}
+  menuItemNode = ${this.menuItemNode}
+  rootline = ${q(this.currentDocumentNode).parents().get()}
+  entryLevel = 0
+
+  renderer = Neos.Fusion:DataStructure {
+    @if.renderHiddenOrVisible = ${!q(props.menuItemNode).property('_hiddenInIndex') || props.renderHiddenInIndex}
```

Comment:
Switch these two. If `renderHiddenInIndex` is set to `true`, it can be rendered faster as there is no need for checking the property